README.md

```diff
@@ -1,22 +1,120 @@
 
-This might take a while...
+---
+
+## Table of Contents
+
+- [Project Overview](#project-overview)
+- [Features](#features)
+- [Architecture](#architecture)
+- [Getting Started](#getting-started)
+- [Development Guide](#development-guide)
+- [Planned Features & Roadmap](#planned-features--roadmap)
+- [Known Issues](#known-issues)
+- [Feedback & Contributions](#feedback--contributions)
+- [Credits](#credits)
+- [License](#license)
+
+---
+
+## Project Overview
+
+**PartBot** aims to provide a powerful and flexible platform for running bots on [Pokemon Showdown](https://pokemonshowdown.com/) and similar systems. The codebase is being rebuilt from scratch to ensure best practices, clear structure, and a solid foundation for future growth. This repository is actively developed and welcomes contributions from the community.
+
+## Features
+
+- **TypeScript-first:** Strongly typed codebase for reliability and maintainability.
+- **Modular Design:** Add, remove, or update components with minimal friction.
+- **Pokémon Showdown Integration:** Automated room management, moderation, and custom features.
+- **Web Module:** Built-in CDN and file hosting, deployable via NGINX with configurable scripts.
+- **Extensible Plugin System:** (Planned) Easily develop and plug in new features or commands.
+- **Community Driven:** Open to suggestions, feedback, and pull requests.
+
+## Architecture
+
+- **Core:** The main logic for bot operations, communication, and orchestration.
+- **Web:** Handles networking, CDN, file hosting, and API endpoints. (See [`src/web/README.md`](src/web/README.md))
+- **Integrations:** Support for third-party systems and APIs.
+- **Frontend:** (Planned) A modern web dashboard for management and analytics.
+
+## Getting Started
+
+### Prerequisites
+
+- [Node.js](https://nodejs.org/) (recommended LTS version)
+- [npm](https://www.npmjs.com/) or [yarn](https://yarnpkg.com/)
+- (Optional) [Docker](https://www.docker.com/) for containerized deployments
+
+### Installation
+
+```bash
+git clone https://github.com/PartMan7/PartBot.git
+cd PartBot
+npm install
+```
+
+### Configuration
+
+Refer to the documentation for each module, especially if customizing NGINX or deploying web features. Environment variables and sensitive data should be stored securely; see `.env.example` if provided.
+
+### Running Locally
+
+```bash
+npm run dev      # Start in development mode
+npm run build    # Build for production
+npm start        # Run production build
+```
+
+---
+
+## Development Guide
+
+- Follow modern JavaScript/TypeScript best practices.
+- Test code before submitting pull requests.
+- Review and adhere to [CONTRIBUTING.md](CONTRIBUTING.md).
+- For web module specifics, see [`src/web/README.md`](src/web/README.md).
```

Comment:
There's no such thing as a 'web module'; `@/web` is almost exclusively for PartBot APIs, custom pages (which wouldn't be added by others), or game replays.